Assets/Scripts/Framework/Gameplay/Timer/TimerUpdater.cs

```diff
@@ -0,0 +1,135 @@
+        private float _chaosSpeedMultiplierThreshold = 60f;
+
+        [SerializeField]
+        private float _chaosSpeedMultiplier = 1.5f;
+
+        [SerializeField]
+        private List<int> _arrivedStations = new();
+
+        /// <summary>
+        /// A getter to get the ChaosSpeedMultiplierThreshold.
+        /// </summary>
+        public float ChaosSpeedMultiplierThreshold => _chaosSpeedMultiplierThreshold;
+
+        /// <summary>
+        /// A getter that returns the seconds that are left on the timer.
+        /// </summary>
+        public float TimeRemaining => _timer.Remaining;
+
+        /// <summary>
+        /// An action that broadcasts when the timer is updated and the new time that comes with it.
+        /// </summary>
+        public Action<float> OnTimerUpdated;
+
+        /// <summary>
+        /// An action that broadcasts when the timer is completed.
+        /// </summary>
+        public Action OnTimerCompleted;
+
+        [SyncObject]
+        private readonly SyncTimer _timer = new();
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        public override void OnStartServer()
+        {
+            base.OnStartServer();
+
+            _timer.StartTimer(_baseTime);
+
+            TrainStationController.Instance.OnParkStateChanged += OnStationArrival;
+        }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        public override void OnStopServer()
+        {
+            base.OnStopServer();
+
+            _timer.StopTimer();
+
+            if (TrainStationController.Instance != null)
+                TrainStationController.Instance.OnParkStateChanged -= OnStationArrival;
+        }
+
+        /// <summary>
+        /// A function that updates the time when the train arrives at a station.
+        /// </summary>
+        private void OnStationArrival(bool isParked)
+        {
+            if (_timer.Paused || !isParked)
+                return;
+
+            int stationIndex = StationManager.Instance.GetNearestStationIndex
+                (
+                    TrainStationController.Instance.CurrentTrainLocation, 
+                    out _
+                );
+
+            if (_arrivedStations.Contains(stationIndex))
+                return;
+
+            _arrivedStations.Add(stationIndex);
+
+            _timer.StartTimer(_timer.Remaining + _stationArrivalTimeBonus);
+
+            OnTimerUpdated?.Invoke(_timer.Remaining);
+        }
+
+        private void Update()
+        {
+            if (_timer.Paused)
+                return;
+
+            UpdateTimer();
+        }
+
+        private void UpdateTimer()
+        {
+            _timer.Update(Time.deltaTime);
+            OnTimerUpdated?.Invoke(_timer.Remaining);
+
+            if (_timer.Remaining <= 0)
+                OnTimerCompleted?.Invoke();
+
+            if (_timer.Remaining <= _chaosSpeedMultiplierThreshold)
+                ApplyChaosMultiplier();
+        }
```

Comment:
Is the chaos multiplier also disabled when the players gets additional time from station arrivals?

Reply:
Resolved.